app/controllers/notes_controller.rb

```diff
@@ -13,50 +25,118 @@ def create
-      end
+    if @note.save
+      record_deed
+      render json: {
+        html: render_to_string(partial: 'note.html', locals: { note: @note }, formats: [:html]),
+        flash: render_to_string(partial: 'shared/flash', locals: { type: 'notice', message: t('.note_has_been_created') },
+                                formats: [:html])
+      }, status: :created
+    else
+      render json: {
+        errors: @note.errors.full_messages,
+        flash: render_to_string(partial: 'shared/flash', locals: { type: 'error', message: t('.error_creating_note') })
+      }, status: :unprocessable_entity
     end
   end
 
   def update
     @note = Note.find(params[:id])
-    respond_to do |format|
-      if @note.update(note_params)
-        note_body = sanitize(@note.body, tags: %w(strong b em i a), attributes: %w(href))
 
-        format.json { render json: { html: simple_format(note_body) }, status: :ok }
-        format.html { redirect_back fallback_location: @note, notice: t('.note_has_been_updated') }
-      else
-        format.json { render json: @note.errors.full_messages, status: :unprocessable_entity }
-        format.html { redirect_back fallback_location: @note, flash: { error: t('.error_updating_note') } }
-      end
+    ajax_redirect_to root_path and return unless user_signed_in? && @note.user == current_user
+
+    if @note.update(note_params)
+      note_body = sanitize(@note.body, tags: %w(strong b em i a), attributes: %w(href))
+
+      render json: {
+        html: simple_format(note_body),
+        flash: render_to_string(partial: 'shared/flash', locals: { type: 'notice', message: t('.note_has_been_updated') },
+                                formats: [:html])
+      }, status: :ok
+    else
+      render json: {
+        errors: @note.errors.full_messages,
+        flash: render_to_string(partial: 'shared/flash', locals: { type: 'error', message: t('.error_updating_note') })
+      }, status: :unprocessable_entity
     end
   end
 
   def destroy
     @note = Note.find(params[:id])
-    @note.destroy
-    respond_to do |format|
-      format.json { head :no_content }
-      format.html { redirect_back fallback_location: root_path, notice: t('.note_has_been_deleted') }
+
+    unless user_signed_in? && (@note.user == current_user || current_user.like_owner?(@note.work))
+      ajax_redirect_to root_path and return
     end
+
+    @note.destroy
+
+    render json: {
+      flash: render_to_string(partial: 'shared/flash', locals: { type: 'notice', message: t('.note_has_been_deleted') })
+    }
   end
 
-  def edit
-    @note = Note.find(params[:id])
+  def discussions
+    @pages = @collection.pages
+                        .where.not(last_note_updated_at: nil)
+                        .reorder(last_note_updated_at: :desc)
+                        .paginate(page: params[:page], per_page: PAGES_PER_SCREEN)
   end
 
-  def show
-    @note = Note.find(params[:id])
+  private
+
+  def filtered_notes
```

Comment:
For filtering/sorting actions, as much as possible we should do it server-side and as efficient as possible (avoiding n+1 queries if possible)

I think this is the simplest I can think of in order to have the notes table searchable by all columns. But if possible I'd like for us to just limit the search to note titles maybe? But if we really need to search with all available columns then this should work.